metadataserver/inject/inject_linux.go

```diff
@@ -338,3 +364,59 @@ done:
 
 	return &socket, nil
 }
+
+func getLinkLocalAddr(ctx context.Context, handle *netlink.Handle, link netlink.Link) (*netlink.Addr, error) {
+	t := time.NewTimer(1)
+	defer t.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("Context complete before able to get LL addrs: %w", ctx.Err())
+		case <-t.C:
+			addrs, err := handle.AddrList(link, netlink.FAMILY_V6)
+			if err != nil {
+				return nil, fmt.Errorf("Could not list V6 addrs on %q: %w", link.Attrs().Name, err)
+			}
```

Comment:
Is this function specific to ipv6? 
If so, can you rename this function to `getLinkLocalIPv6Addr` maybe?
I just worry sometimes when I see `addr.IP` and making a mistake thinking that it is v4 when it is actually v6.

Reply:
Right, that makes sense.